### ISSUE TITLE
Both horizontal and vertical scroll?

### ISSUE THREAD

**Issue by user97116** [baumths/flutter_tree_view#53]
I used and not work in CustomScroll, CustomScroll, SingleChildScrollView any solution?

**Comment by baumths** [comment]
Waiting on https://flutter.dev/go/2D-Foundation

**Comment by Hilbert2048** [comment]
This is a feature I really want

![image](https://github.com/baumths/flutter_tree_view/assets/125776109/444a6513-ab4e-42ed-89bc-559b28d3b6b6)

**Comment by julek-kal** [comment]
@baumths fyi - since flutter 3.13 is live I think that 2D scrolling is ready to use.
https://medium.com/flutter/whats-new-in-flutter-3-13-479d9b11df4d

**Comment by igormidev** [comment]
@julek-kal 

Are you sure Flutter 3.13 have everything it takes to implement 2D scrolling in this package? They will still add several things about 2d scroll in flutter... 3.13 is just the start

**Comment by julek-kal** [comment]
@igormidev 
No, I'm not sure. That was just info for the author, cause he mentioned that we need to wait for 2D foundation. Since 3.13 has it, I decided to kindly let him know🙂

**Comment by stephane-archer** [comment]
is there any alternative while waiting for this?

**Comment by baumths** [comment]
I'm afraid I won't be able to add horizontal scroll support to the tree view widgets.

I've played around with the new 2D scrolling API and found out I won't be able to put in the amount of time required to develop this feature because of the complexity involved.

**Comment by baumths** [comment]
> is there any alternative while waiting for this?

The "ugly" way of doing it currently is to set a fixed width to your TreeView:
```dart
@override
Widget build(BuildContext context) {
  return SingleChildScrollView(
    scrollDirection: Axis.horizontal,
    child: SizedBox(
      // width: <some-fixed-number>,
      width: MediaQuery.sizeOf(context).width * 2,
      child: TreeView(
        ...
      ),
    ),
  );
}
```

**Comment by DirtyNative** [comment]
@baumths I dont get the horizontal scroll to work, did you do something else?

**Comment by rubenlop88** [comment]
> > is there any alternative while waiting for this?
> 
> The "ugly" way of doing it currently is to set a fixed width to your TreeView:
> 
> ```dart
> @override
> Widget build(BuildContext context) {
>   return SingleChildScrollView(
>     scrollDirection: Axis.horizontal,
>     child: SizedBox(
>       // width: <some-fixed-number>,
>       width: MediaQuery.sizeOf(context).width * 2,
>       child: TreeView(
>         ...
>       ),
>     ),
>   );
> }
> ```

Thanks. This works just fine.

I calculate the width like this:

```
class Node {
  ...
  int get maxDepth {
    if (isEmpty) {
      return 0;
    }
    return 1 + _children.map((child) => child.maxDepth).reduce(math.max);
  }
}
```


```
  @override
  Widget build(BuildContext context) {
    final itemWidth = 300.0;
    return DefaultIndentGuide(
      guide: const ConnectingLinesGuide(),
      child: Builder(
        builder: (context) {
          final depth =  root.maxDepth;
          final indent = DefaultIndentGuide.of(context).indent;
          final treeWidth = itemWidth + depth * indent;
          return SingleChildScrollView(
            scrollDirection: Axis.horizontal,
            child: Container(
              width: treeWidth,
              child: AnimatedTreeView<Node>(
                ...
                nodeBuilder: (context, entry) => SizedBox(
                  width: itemWidth,
                  child: ...,
                ),
              ),
            ),
          );
        },
      ),
    );
  }
```
